opentracing/src/main/java/opentracing/Tracer.java

```diff
@@ -0,0 +1,95 @@
+/**
+ * Copyright 2016 The OpenTracing Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package opentracing;
+
+
+/**
+ * Tracer is a simple, thin interface for Span creation, and Span propagation into different transport formats.
+ */
+public interface Tracer {
+
+  /**
+   * Create, start, and return a new Span with the given `operationName`.
+   * An optional parent Span can be specified used to incorporate the newly-returned Span into an existing trace.
+   *
+   * <p>Example:
+   * <pre>{@code
+    Tracer tracer = ...
+
+    Span feed = tracer.buildSpan("GetFeed")
+                      .start();
+
+    Span http = tracer.buildSpan("HandleHTTPRequest")
+                      .withParent(feed)
+                      .withTag("user_agent", req.UserAgent)
+                      .withTag("lucky_number", 42)
+                      .start();
+    }</pre>
+   */
+  SpanBuilder buildSpan(String operationName);
+
+  /** Takes two arguments:
+   *    a Span instance, and
+   *    a “carrier” object in which to inject that Span for cross-process propagation.
+   *
+   * A “carrier” object is some sort of http or rpc envelope, for example HeaderGroup (from Apache HttpComponents).
```

Comment:
These required carriers are in https://github.com/opentracing/opentracing-java/pull/11

The non-interoperable i presume you mean is at the "format" level.
I don't see this is a problem as long as it's clearly documented with java that these are the required carriers and are otherwise interoperable.